hack-sound-server

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+import gi
+import logging
+import json
+import os
+import signal
+import uuid
+gi.require_version('GLib', '2.0')  # noqa
+gi.require_version('Gst', '1.0')   # noqa
+gi.require_version('Json', '1.0')  # noqa
+from gi.repository import Gio
+from gi.repository import GLib
+from gi.repository import GObject
+from gi.repository import Gst
+from gi.repository import Json
+
+
+_logger = logging.getLogger(__name__)
+
+
+def get_datadir():
+    return os.path.join(GLib.get_user_data_dir(), "hack-sound-server")
+
+def get_metadata_path():
+    return os.path.join(get_datadir(), "metadata.json")
+
+def get_sounds_dir():
+    return os.path.join(get_datadir(), "sounds")
+
+
+class HackSoundPlayer(GObject.Object):
+    __gsignals__ = {
+        'eos': (GObject.SIGNAL_RUN_FIRST, None, ()),
+        'error': (GObject.SIGNAL_RUN_FIRST, None, (object, str))
```

Comment:
I'm not entirely sure if you can do this with the `__gsignals__` dict but you probably want the type of the first parameter to be `G_TYPE_ERROR`, or in other words `GLib.Error`, instead of `object`.